### PR TITLE
Re-instate IE11 fix

### PR DESCRIPTION
**Pull request by kevindew**
Trello: https://trello.com/c/e1AYHybj/672-put-together-the-copy-and-paste-work-as-a-package-that-can-be-included

D'oh! I somehow lost this in a bad merge.

To try reduce the chances of this happening again I've added a test for
this. By some witchcraft I can mock an import after running other
imports - hopefully that's intended functionality. This allows mocking
the htmlFromHiddenElement function which simplifies this test a lot.

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -1,11 +1,13 @@
 import toGovspeak from './to-govspeak'
 import insertTextAtCursor from 'insert-text-at-cursor'
+import htmlFromHiddenElement from './html-from-hidden-element'
 
 function htmlFromPasteEvent (event) {
-  if (!event.clipboardData) {
-    return
+  if (event.clipboardData) {
```
Comment (benthorner) [inline src/main.js]: It's great we can support this, but I think the "what's it for" aspect has got lost a little - my first question here was "why is there a hidden element"? Although it's documented in the test, I think we could use a couple of question methods, and perhaps a better name for that import, to better explain what's going on to the read.

**Comment by benthorner** [comment]
@kevindew [this](https://jestjs.io/docs/en/es6-class-mocks#calling-jestmock-docs-en-jest-object-jestmockmodulename-factory-options-with-the-module-factory-parameter) seems to explain why the jest technique works

File: src/main.js (hunk 1 above)
Comment (kevindew) [reply src/main.js]: Fair point, we had more comments before and I think they slipped - can imagine it's a bit voodoo.

**Comment by kevindew** [comment]
@benthorner I've added some explanatory comments, thanks for the jest info 👍

File: src/main.js [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
 import insertTextAtCursor from 'insert-text-at-cursor'
+import htmlFromHiddenElement from './html-from-hidden-element'
 
 function htmlFromPasteEvent (event) {
-  if (!event.clipboardData) {
-    return
+  // Modern browsers
+  if (event.clipboardData) {
+    return event.clipboardData.getData('text/html')
+  } else if (window.clipboardData) { // IE propriety property
```
Comment (benthorner) [inline src/main.js]: I think this is 'proprietary' :-).

File: src/main.js [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
+    // As IE doesn't support event.clipboardData we have to use a hack to
+    // access HTML from a paste
+    return htmlFromHiddenElement()
```
Comment (benthorner) [inline src/main.js]: I'm still concerned this is going to be hard to understand for someone coming into this repo for the first time - the question 'why is there a hidden element?' pops up every time I read this, which suggests we could abstract this more cleanly. After reading the file, the hidden element seems to be an implementation detail, so we could just say 'htmlFromPasteEventIE'.

**Comment by kevindew** [comment]
@benthorner this is updated again.

**Pull request merged**